### PR TITLE
Rd 11126 lambda python node tracers sqs json support

### PR DESCRIPTION
**Pull request by sagivoululumigo**
# Features
* Added support for SQS messages sent with the new JSON protocol (see [AWS docs](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-json-faqs.html))

### CODE REVIEW

**Review by nadav3396 — APPROVED** [review]
LGTM

File: src/parsers/aws.ts [hunk 1]
```diff
@@ -122,36 +129,64 @@ export const eventBridgeParser = (requestData, responseData) => {
...
-        'MessageId',
-      ],
-      undefined
-    );
+  let awsServiceData = {};
+
+  // Note: Currently json protocol is new and not commonly used, so the default case is XML.
+  // In the future when JSON usage is default you my want to switch so JSON is the default and XML
+  // protocol is specifically checked for
+  if (
```
Comment (doriaviram) [inline src/parsers/aws.ts:140]: Make sure we know to extract the `resourceName` in JSON also from other SQS methods like [sendMessageBatch](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/SQS.html#sendMessageBatch-property)

File: src/parsers/aws.ts [hunk 2]
```diff
@@ -122,36 +129,64 @@ export const eventBridgeParser = (requestData, responseData) => {
 };
 
 export const sqsParser = (requestData, responseData) => {
-  const { body: reqBody } = requestData;
+  const { body: reqBody } = requestData || {};
```
Comment (doriaviram) [inline src/parsers/aws.ts:132]: Let's split this function into 2, XML & JSON, to make it more readable

File: src/utils.ts [hunk 3]
```diff
@@ -268,6 +268,19 @@ export const safeGet = (obj, arr, dflt = null) => {
...
 };
+/**
+ * Finds a value from object with case-insensitive match of keys.
+ * If multiple matching keys are found, the first is returned
+ *
+ * @param obj Object to get value from
+ * @param key The key to search (can be upper / lower case or a mix of the two, it doesn't matter)
+ * @param dflt Default value if no match is found
+ */
+export const caseInsensitiveGet = (obj, key, dflt = null) => {
```
Comment (doriaviram) [inline src/utils.ts:279]: Add types

**Review by doriaviram — CHANGES_REQUESTED** [review]

File: src/parsers/aws.ts (hunk 1 above)
Comment (sagivoululumigo) [reply src/parsers/aws.ts:140]: I tested for this in the ```sqsParser -> happy flow JSON format```, the test is run on a single message & a batch message & the message id & resource name are tested

File: src/parsers/aws.ts (hunk 2 above)
Comment (sagivoululumigo) [reply src/parsers/aws.ts:149]: Fixed

File: src/parsers/aws.ts (hunk 1 above)
Comment (sagivoululumigo) [reply src/parsers/aws.ts:191]: Done, fixed it

**Review by doriaviram — APPROVED** [review]

**Pull request merged**